app/queue/notifications.py

```diff
@@ -0,0 +1,95 @@
+from datetime import datetime
+from datetime import timezone
+from enum import Enum
+
+from marshmallow import fields
+from marshmallow import Schema as MarshmallowSchema
+from marshmallow import validate as marshmallow_validate
+
+from app.logging import threadctx
+from app.models import CanonicalFactsSchema
+from app.queue.events import hostname
+from app.queue.metrics import notification_event_serialization_time
+
+NotificationType = Enum("NotificationType", ("validation_error"))
+EventSeverity = Enum("EventSeverity", ("warning", "error", "critical"))
+
+
+# Schemas
+class HostValidationErrorSchema(MarshmallowSchema):
+    code = fields.Str(required=True)
+    message = fields.Str(required=True, validate=marshmallow_validate.Length(max=1024))
+    stack_trace = fields.Str()
+    severity = fields.Str(required=True, validate=marshmallow_validate.OneOf(EventSeverity))
+
+
+class ErrorPayloadSchema(MarshmallowSchema):
+    canonical_facts: fields.Nested(CanonicalFactsSchema())
+    error = fields.Nested(HostValidationErrorSchema())
+
+
+class HostValidationErrorMetadataSchema(MarshmallowSchema):
+    metadata = fields.Dict(validate=marshmallow_validate.Equal({}))
+    payload = fields.Nested(ErrorPayloadSchema())
+
+
+class HostValidationErrorNotificationEvent(MarshmallowSchema):
+    id: fields.UUID(required=True)
+    version = fields.Str(required=True, validate=marshmallow_validate.Length(max=10))
+    bundle = fields.Str(required=True, validate=marshmallow_validate.Equal("rhel"))
+    application = fields.Str(required=True, validate=marshmallow_validate.Equal("inventory"))
+    event_type = fields.Str(required=True, validate=marshmallow_validate.Length(max=256))
+    timestamp = fields.DateTime(required=True, format="iso8601")
+    account_id = fields.Str(required=True, validate=marshmallow_validate.Length(min=0, max=36))
+    org_id = fields.Str(validate=marshmallow_validate.Length(min=0, max=36))
+    context = fields.Dict()
+    events = fields.Nested(HostValidationErrorMetadataSchema())
+
+
+def notification_message_headers(event_type: NotificationType, rh_message_id: bytearray = None):
+    return {
+        "event_type": event_type.name,
+        "request_id": threadctx.request_id,
+        "producer": hostname(),
+        "rh-message-id": rh_message_id,
+    }
+
+
+def host_validation_error_event(notification_type, host, detail, stack_trace=None):
+    validation_error_event = {
+        "version": "v1.0.0",
+        "bundle": "rhel",
+        "application": "inventory",
+        "event_type": notification_type.name.replace("_", "-"),
+        "timestamp": datetime.now(timezone.utc),
+        "account_id": host["account_id"],
+        "org_id": host["org_id"] if host["org_id"] else None,
```

Comment:
to prepare for account_id going away and org_id coming in painlessly

```suggestion
        "account_id": host.get("account_id"),
        "org_id": host.get("org_id"),
```